k2/torch/csrc/torch_api.cu

```diff
@@ -74,13 +74,13 @@ FsaClassPtr LoadFsaClass(const std::string &filename,
   return std::make_shared<FsaClass>(LoadFsa(filename, map_location));
 }
 
-std::vector<std::vector<int32_t>> Decode(torch::Tensor log_softmax_out,
-                                         torch::Tensor log_softmax_out_lens,
-                                         FsaClassPtr decoding_graph,
-                                         float search_beam, float output_beam,
-                                         int32_t min_activate_states,
-                                         int32_t max_activate_states,
-                                         int32_t subsampling_factor) {
+FsaClassPtr GetLattice(torch::Tensor log_softmax_out,
```

Comment:
I suggest that we provide only a single method `Decode()` to give the results directly.

In the current approach, users have to call `BestPath` on the returned lattice and that is the only function that users can call for the returned lattice. 

That is an implementation detail and we can hide it from the users.

Reply:
> I suggest that we provide only a single method `Decode()` to give the results directly.
> 
> In the current approach, users have to call `BestPath` on the returned lattice and that is the only function that users can call for the returned lattice.
> 
> That is an implementation detail and we can hide it from the users.

I thought we can use this lattice to do rescoring, we can implement `one_best_decoding` and `ngram_rescoring` in sherpa.

Reply:
I think it is no need to hide lattice.